Marlin/src/pins/lpc1769/pins_BTT_SKR_E3_TURBO.h

```diff
@@ -0,0 +1,266 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#ifndef BOARD_INFO_NAME
+  #define BOARD_INFO_NAME "BTT SKR E3 Turbo"
+#endif
+
+// Onboard I2C EEPROM
+#define I2C_EEPROM
+#define MARLIN_EEPROM_SIZE                0x1000  // 4KB (AT24C32)
+
+//
+// Servos
+//
+#define SERVO0_PIN                         P1_23
+
+//
+// TMC StallGuard DIAG pins
+//
+#define X_DIAG_PIN                         P1_29  // X-STOP
+#define Y_DIAG_PIN                         P1_28  // Y-STOP
+#define Z_DIAG_PIN                         P1_27  // Z-STOP
+#define E0_DIAG_PIN                        P1_26  // E0DET
+#define E1_DIAG_PIN                        P1_25  // E1DET
+
+//
+// Limit Switches
+#define X_STOP_PIN                    X_DIAG_PIN
+#define Y_STOP_PIN                    Y_DIAG_PIN
+#define Z_STOP_PIN                    Z_DIAG_PIN
+
+//
+// Z Probe
+//
+#ifndef Z_MIN_PROBE_PIN
+  #define Z_MIN_PROBE_PIN                  P1_22
+#endif
+
+//
+// Filament Runout Sensor
+//
+#define FIL_RUNOUT_PIN                     P1_26  // E0DET
+#define FIL_RUNOUT2_PIN                    P1_25  // E1DET
+
+//
+// Power Supply Control
+//
+#ifndef PS_ON_PIN
+  #define PS_ON_PIN                        P1_21
+#endif
+
+// LED driving pin
+#define NEOPIXEL_PIN                       P1_24
+
+//
+// Power Loss Detection
+//
+#ifndef POWER_LOSS_PIN
+  #define POWER_LOSS_PIN                   P1_20  // PWRDET
+#endif
+
+//
+// Steppers
+//
+#define X_STEP_PIN                         P1_04
+#define X_DIR_PIN                          P1_08
+#define X_ENABLE_PIN                       P1_00
+#ifndef X_CS_PIN
+  #define X_CS_PIN                         P1_01
+#endif
+
+#define Y_STEP_PIN                         P1_14
+#define Y_DIR_PIN                          P1_15
+#define Y_ENABLE_PIN                       P1_09
+#ifndef Y_CS_PIN
+  #define Y_CS_PIN                         P1_10
+#endif
+
+#define Z_STEP_PIN                         P4_29
+#define Z_DIR_PIN                          P4_28
+#define Z_ENABLE_PIN                       P1_16
+#ifndef Z_CS_PIN
+  #define Z_CS_PIN                         P1_17
+#endif
+
+#define E0_STEP_PIN                        P2_06
+#define E0_DIR_PIN                         P2_07
+#define E0_ENABLE_PIN                      P0_04
+#ifndef E0_CS_PIN
+  #define E0_CS_PIN                        P0_05
+#endif
+
+#define E1_STEP_PIN                        P2_11
+#define E1_DIR_PIN                         P2_12
+#define E1_ENABLE_PIN                      P0_21
+#ifndef E1_CS_PIN
+  #define E1_CS_PIN                        P0_22
+#endif
+
+#if HAS_TMC_UART
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   * Hardware serial communication ports.
```

Comment:
Personal opinion here.

I think when board have built-in serial wiring, we should just define whatever is physically present (Hardware or Software serial), and avoid even mentioning the other.

Reply:
Fair enough. I was just using the boilerplate text from other pins files.